graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/CodeGen.kt

```diff
@@ -409,8 +409,16 @@ fun List<FieldDefinition>.filterIncludedInConfig(definitionName: String, config:
     }
 }
 
-fun ObjectTypeDefinition.shouldSkip(): Boolean {
-    return this.directives.any { it.name == "skipcodegen" }
+fun ObjectTypeDefinition.shouldSkip(config: CodeGenConfig): Boolean {
+    return this.directives.any { it.name == "skipcodegen" } || config.typeMapping.containsKey(this.name)
+}
+
+fun InterfaceTypeDefinition.shouldSkip(config: CodeGenConfig): Boolean {
+    return this.directives.any { it.name == "skipcodegen" } || config.typeMapping.containsKey(this.name)
+}
+
+fun UnionTypeDefinition.shouldSkip(config: CodeGenConfig): Boolean {
```

Comment:
This duplication across the definition types strikes me as a design flaw in the type abstraction - shouldn't there be a common component to the type definitions that can more elegantly express the shared logic?

Reply:
I agree, but these types come from graphql-java, and aren't "ours".